contracts/foundation/application.go

```diff
@@ -25,4 +26,12 @@ type Application interface {
 	PublicPath(path string) string
 	// Publishes register the given paths to be published by the "vendor:publish" command.
 	Publishes(packageName string, paths map[string]string, groups ...string)
+	// GetLocale get the current application locale.
+	GetLocale(ctx http.Context) string
+	// SetLocale set the current application locale.
+	SetLocale(ctx http.Context, locale string)
+	// Version gets the version number of the application.
+	Version() string
+	// IsLocale get the current application locale.
+	IsLocale(ctx http.Context, locale string) bool
```

Comment:
```suggestion
	// GetLocale get the current application locale.
	GetLocale(ctx context.Context) string
	// SetLocale set the current application locale.
	SetLocale(ctx context.Context, locale string) context.Context
	SetLocaleByHttp(ctx http.Context, locale string)
	// Version gets the version number of the application.
	Version() string
	// IsLocale get the current application locale.
	IsLocale(ctx context.Context, locale string) bool
```

Reply:
Why do we need `SetLocaleByHttp` function?

Reply:
We need a function to deal with `http.Context`.

Reply:
But it can be considered as `context.Context`

Reply:
Imagine, if `http.Context` is passed to `SetLocale(ctx context.Context, locale string) context.Context`, how can we set the `context.Context` return value to `http.Context`?

Reply:
Okay